harmonica/datasets/sample_data.py

```diff
@@ -140,6 +140,30 @@ def fetch_rio_magnetic():
     return pd.read_csv(POOCH.fetch("rio-magnetic.csv.xz"), compression="xz")
 
 
+def fetch_gb_magnetic():
+    """
+    Fetch total-field magnetic anomaly data of the United Kingdom.
+
+    These data are a complete airborne survey of the entire United Kingdom
+    conducted between 1955 and 1965. The data are made available by the
+    British Geological Survey (BGS) through their `geophysical data portal
+    <https://www.bgs.ac.uk/products/geophysics/aeromagneticRegional.html>`__.
```

Comment:
```suggestion
    <https://www.bgs.ac.uk/products/geophysics/aeromagneticRegional.html>`__.
    
    License: `Open Government License 
    <http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/>`__
```

Reply:
We're trying to standardize this in our sample datasets. 

@santisoler what do you think of having this `License:` section in the docstring?

Reply:
Sure! I think we should add the license both on the gallery example and on the docstring of the function that fetches the dataset. Besides, I think adding the name of the license and a link to the original license text would be enough.